src/main/java/com/epam/processor/DataProcessor.java

```diff
@@ -1,123 +1,240 @@
+			} else if (size > countList.get(1)) {
+				list.add(1, key);
+				countList.add(1, size);
+			} else if (size > countList.get(2)) {
+				list.add(2, key);
+				countList.add(2, size);
+			}
+			if (list.size() == 4) {
+				list.remove(3);
+			}
+		}
+		return list;
+	}
+
+	/**
+	 * return a multimap where key is a district authority and values are
+	 * accident ids ex: authority1 -> id1, id2, id3 authority2 -> id4, id5
+	 * 
+	 * @return
+	 */
+	public Multimap<String, String> getAccidentIdsGroupedByAuthority7() {
+		Multimap<String, String> multiMap = HashMultimap.create();
+		long count = 0L;
+		for (RoadAccident roadAccident : roadAccidentList) {
+			multiMap.put(roadAccident.getDistrictAuthority(),
+					roadAccident.getAccidentId() + count++);
+		}
+		return multiMap;
+	}
+
+	// Now let's do same tasks but now with streaming api
+
+	public RoadAccident getAccidentByIndex(String index) {
+		RoadAccident roadAccident = roadAccidentList.stream()
+				.filter(ra -> ra.getAccidentId().equals(index)).findFirst()
+				.orElse(null);
+		return roadAccident;
+	}
+
+	/**
+	 * filter list by longtitude and latitude fields
+	 * 
+	 * @param minLongitude
+	 * @param maxLongitude
+	 * @param minLatitude
+	 * @param maxLatitude
+	 * @return
+	 */
+	public Collection<RoadAccident> getAccidentsByLocation(float minLongitude,
+			float maxLongitude, float minLatitude, float maxLatitude) {
+		return roadAccidentList
+				.stream()
+				.filter(roadAccident -> roadAccident.getLongitude() >= minLongitude
+						&& roadAccident.getLongitude() <= maxLongitude)
+				.filter(roadAccident -> roadAccident.getLatitude() >= minLatitude
+						&& roadAccident.getLatitude() <= maxLatitude)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * find the weather conditions which caused max number of incidents
+	 * 
+	 * @return
+	 */
+	public List<String> getTopThreeWeatherCondition() {
+		List<String> list = new ArrayList<>();
+		Map<String, Long> map = roadAccidentList
+				.stream()
+				.map(RoadAccident::getWeatherConditions)
+				.collect(
+						Collectors.groupingBy(item -> item,
+								Collectors.counting()));
+
+		return map
+				.entrySet()
+				.stream()
+				.sorted((val1, val2) -> val2.getValue().compareTo(
+						val1.getValue())).map(Map.Entry::getKey).limit(3)
+				.collect(Collectors.toList());
+	}
+
+	/**
+	 * count incidents by road surface conditions
+	 * 
+	 * @return
+	 */
+	public Map<String, Long> getCountByRoadSurfaceCondition() {
+		Map<String, Long> map = new HashMap<>();
+
+		long dry = roadAccidentList
+				.stream()
+				.filter(ra -> ra.getRoadSurfaceConditions().equals("5")
+						|| ra.getRoadSurfaceConditions()
+								.equalsIgnoreCase("dry")).count();
+		long wet = roadAccidentList
+				.stream()
+				.filter(ra -> ra.getRoadSurfaceConditions().equals("2")
+						|| ra.getRoadSurfaceConditions()
+								.equalsIgnoreCase("wet")).count();
+
+		map.put("Dry", dry);
+		map.put("Wet", wet);
+		return map;
+	}

```

Comment:
@zero1608 
Your understanding is not correct on this. Please read again & change the code accordingly.